### ISSUE TITLE
The connection does not work after opensearch is replaced with datanode.

### ISSUE THREAD

**Issue by Yaminyam** [Graylog2/docker-compose#42]
org.graylog2.storage.versionprobe.VersionProbe - OpenSearch/Elasticsearch is not available. Retry #3

I recently noticed that opensearch has been replaced by datanode.
However, since then, I keep getting the error that opensearch is not found in graylog.
Is there any way to fix this?

**Comment by janheise** [comment]
@Yaminyam Hi, can you give more info about your setup? Depending on your setup, you might only need to remove the `elasticsearch_hosts` setting in `server.conf`

**Comment by Yaminyam** [comment]
@janheise Thank you for the quick response!
I set up the server using only the cluster version of docker-compose in the repo without any additional settings!
I confirmed that the server was running normally until last week, but it is not running now.

**Comment by janheise** [comment]
@Yaminyam so you're only doing a `docker compose up` again and it's not working? If this is the case, can you try to first try to start the mongo and datanodes (e.g. `docker compose up -d mongodb1 mongodb2 mongodb3 datanode1 datanode2 datenode3`) and then later add the graylog nodes with `docker compose up graylog1` and see what happens?

**Comment by Yaminyam** [comment]
I just deleted all volumes and the opensearch related errors were resolved.
However, while looking at the volume, I discovered another problem.
It seems that the settings in .env are not reflected properly, but there seems to be a problem with the volume mount.
I created an .env file, but when I go into the actual volumes folder and check, there is no value in it.
<img width="771" alt="image" src="https://github.com/Graylog2/docker-compose/assets/31057849/f4fd2ab5-a76e-4875-aeb9-d5db7c8eae22">
cluster-graylog1-1   | java.util.NoSuchElementException: null

**Comment by Yaminyam** [comment]
Looking at the docker-compose file, it looks like the graylog.conf file will be included in the journal or data volumes. Is this correct that volumes with unknown names are created and stored there?
<img width="1007" alt="image" src="https://github.com/Graylog2/docker-compose/assets/31057849/fd0159c1-3e3b-42f7-8f43-5de259fb4bc2">

**Comment by janheise** [comment]
If the problem reoccurs, please let me know. Please post longer stacktraces `cluster-graylog1-1 | java.util.NoSuchElementException: null` probably can't be tracked down unless it's a real obvious error ;-)

The `graylog.conf` files inside docker are not getting modified by the `.env` entries but are set in the environment of the container. Can you give me a few more hints on what you want to achieve and what's not working?

**Comment by Yaminyam** [comment]
cluster-graylog1-1   | 2023-11-20 15:29:37,814 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
cluster-graylog1-1   | java.util.NoSuchElementException: null
cluster-graylog1-1   | 	at java.util.StringTokenizer.nextToken(Unknown Source) ~[?:?]
cluster-graylog1-1   | 	at org.graylog2.bootstrap.preflight.web.BasicAuthFilter.filter(BasicAuthFilter.java:68) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:108) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:44) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:173) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:247) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [graylog.jar:?]
cluster-graylog1-1   | 	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.lang.Thread.run(Unknown Source) [?:?]
cluster-graylog1-1   | 2023-11-20 15:29:38,435 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
cluster-graylog1-1   | java.util.NoSuchElementException: null
cluster-graylog1-1   | 	at java.util.StringTokenizer.nextToken(Unknown Source) ~[?:?]
cluster-graylog1-1   | 	at org.graylog2.bootstrap.preflight.web.BasicAuthFilter.filter(BasicAuthFilter.java:68) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:108) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:44) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:173) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:247) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [graylog.jar:?]
cluster-graylog1-1   | 	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.lang.Thread.run(Unknown Source) [?:?]
cluster-graylog1-1   | 2023-11-20 15:29:38,657 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
cluster-graylog1-1   | java.util.NoSuchElementException: null
cluster-graylog1-1   | 	at java.util.StringTokenizer.nextToken(Unknown Source) ~[?:?]
cluster-graylog1-1   | 	at org.graylog2.bootstrap.preflight.web.BasicAuthFilter.filter(BasicAuthFilter.java:68) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:108) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:44) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:173) ~[graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:247) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [graylog.jar:?]
cluster-graylog1-1   | 	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [graylog.jar:?]
cluster-graylog1-1   | 	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
cluster-graylog1-1   | 	at java.lang.Thread.run(Unknown Source) [?:?]

**Comment by Yaminyam** [comment]
This is the full text of the error
<img width="811" alt="image" src="https://github.com/Graylog2/docker-compose/assets/31057849/5dffd001-8c05-4f3c-ab94-82c370bb2d23">

**Comment by Yaminyam** [comment]
And you said that graylog.conf is created with the container settings. If you mean the environment settings value of docker-compose, isn't it correct that the value is set by reading the .env file?

**Comment by janheise** [comment]
`And you said that graylog.conf is created with the container settings.` - this is not what I said. I said that your `.env` is not modifying a `graylog.conf` file, but graylog reads these variables from the environment on startup - so you don't see any changes in eht screenshot you showed.

Also, it looks as if you're trying to connect to graylog with incomplete credentials? Can you use a `private` window in your browser to connect to `localhost:9000`? do you get the basic auth login window of the browser? What do you enter? the preflight credentials that have been created and printed in the logs of `graylog1`?

**Comment by Yaminyam** [comment]
The same error occurs even if you run it in a secret tab.
The reason I assumed that .env was not reflected was because when logging in from the browser, I had to enter a space instead of the password I entered in .env to log in.

**Comment by janheise** [comment]
Did you read/follow our instructions at https://go2docs.graylog.org/5-2/setting_up_graylog/graylog_data_node_getting_started.htm?tocpath=Setting%20up%20Graylog%7CGraylog%20Data%20Node%7C_____1 ?

**Comment by janheise** [comment]
On your first login to the Preflight UI, you have to use automatically generated credentials, not the credentials you entered in `.env`. Entering a space is definitely not correct and you're not logged in - but instead see the error. If you followed our docs, please let me know where they are unclear so that we can fix/improve them.

**Comment by Yaminyam** [comment]
Thank you. When I checked the log of the graylog1 node, it was telling me the password to log in, not the password I entered!
I was able to log in normally with that password!

**Comment by janheise** [comment]
Excellent! So, do we have to state that more prominent in the docs? Was this an oversight? How can we improve it?

**Comment by Yaminyam** [comment]
no! Your docs is very good!
The problem was that I did not read the document, so I did not realize that I had to use a different password than the password set in env when setting up init.

I only looked at the repo and set it up, so I missed the document.
It would be even better if the document was introduced in the readme!

**Comment by janheise** [comment]
Thank you for the feedback, we'll add that part to the readme.

**Comment by Yaminyam** [comment]
Current readme
   <your password from GRAYLOG_ROOT_PASSWORD_SHA2>
It seems to be confusing even more.

**Comment by ralfbergs** [comment]
I essentially suffered from the same problem, and it took me about 2 hrs to come here, which finally gave me the clue.

My specific problem was that I was installing a more recent version of Graylog, after discarding my initial installation. What remained, however, was my Apache frontend to make Graylog accessible from the internet. I have a basic auth configured in Apache to protect Graylog from unauthorized people -- and *that* was what was giving me the headaches.

When I finally connected to port 9000 in the Docker via an ssh tunnel that I set up, I was able to understand what was going on (and what was going wrong) -- because then I saw the basic auth that Graylog itself produced. And when I entered the credentials found in the container log ("admin" and the auto-generated password), I could log on and initiate the installation. :-)

So, thanks for this helpful dialog here.
